client/pages/offer-thread.jsx

```diff
@@ -62,30 +86,33 @@ class OfferThread extends React.Component {
     })
       .then(response => response.json())
       .then(reply => {
-
       })
       .catch(err => console.error(err));
     this.setState({ reply: '' });
   }
 
+  componentDidUpdate() {
+    this.scrollToBottom();
+  }
+
+  scrollToBottom() {
+    this.messagesEnd && this.messagesEnd.scrollIntoView({ behavior: 'smooth' });
+  }
+
```

Comment:
You'll need to add a `componentWillUnmount() {` to disconnect the socket when the user leaves this screen. Otherwise, you'll have a memory leak.

https://reactjs.org/docs/react-component.html#componentwillunmount